fictadvisor-api/src/v2/api/services/TeacherService.ts

```diff
@@ -305,26 +302,22 @@ export class TeacherService {
     });
 
     if (!dbTeacher) {
-      throw new InvalidEntityIdException('subject');
+      throw new InvalidEntityIdException('Subject');
     }
 
-    const { disciplineTeachers, ...teacher } = dbTeacher;
+    const { disciplineTeachers } = dbTeacher;
 
     const roles = this.disciplineTeacherMapper.getRolesBySubject(disciplineTeachers as unknown as DbDisciplineTeacher[], subjectId);
-    const { disciplines, ...subject } = await this.subjectRepository.findById(subjectId);
+    const subject = await this.subjectRepository.findById(subjectId);
     const contacts = await this.contactRepository.getAllContacts(teacherId);
-    const data = {
-      subjectId,
-    };
-    const marks = await this.getMarks(teacherId, data);
-    const rating = this.getRating(marks);
+    const marks = await this.getMarks(teacherId, { subjectId });
 
     return {
-      ...teacher,
-      subject,
+      ...this.teacherMapper.getTeacher(dbTeacher),
+      subject: this.disciplineTeacherMapper.getSubject(subject),
       roles,
       contacts,
-      rating,
+      rating: this.getRating(marks),
```

Comment:
Since the rating is already in the teacher entity, so it's calculated in the mapper (see getSubject method). So delete it.